denote.el

```diff
@@ -1015,7 +1015,8 @@ are not backups."
    #'expand-file-name
    (seq-filter
     (lambda (file)
-      (and (denote-file-has-identifier-p file)
+      (and (file-regular-p file)
+           (denote-file-has-identifier-p file)
```

Comment:
I don't think `file-regular-p` should be part of `denote-file-has-identifier-p` because I do not think the latter should be doing any extra work or checking. And actually, `denote-file-has-identifier-p` may be called with a file name that does not (yet) exists. This scenario is possible.